gym/spaces/box.py

```diff
@@ -22,67 +24,71 @@ class Box(Space):
 
         # determine shape if it isn't provided directly
         if shape is not None:
             shape = tuple(shape)
-            assert (
-                np.isscalar(low) or low.shape == shape
-            ), "low.shape doesn't match provided shape"
-            assert (
-                np.isscalar(high) or high.shape == shape
-            ), "high.shape doesn't match provided shape"
         elif not np.isscalar(low):
-            shape = low.shape
-            assert (
-                np.isscalar(high) or high.shape == shape
-            ), "high.shape doesn't match low.shape"
+            shape = low.shape  # type: ignore
         elif not np.isscalar(high):
-            shape = high.shape
-            assert (
-                np.isscalar(low) or low.shape == shape
-            ), "low.shape doesn't match high.shape"
+            shape = high.shape  # type: ignore
         else:
             raise ValueError(
                 "shape must be provided or inferred from the shapes of low or high"
             )
+        assert isinstance(shape, tuple)
 
         if np.isscalar(low):
             low = np.full(shape, low, dtype=dtype)
 
         if np.isscalar(high):
             high = np.full(shape, high, dtype=dtype)
 
-        self._shape = shape
-        self.low = low
-        self.high = high
+        assert isinstance(low, np.ndarray)
+        assert low.shape == shape, "low.shape doesn't match provided shape"
+        assert isinstance(high, np.ndarray)
+        assert high.shape == shape, "high.shape doesn't match provided shape"
+
+        self._shape: Tuple[int, ...] = shape
 
-        def _get_precision(dtype):
+        def _get_precision(dtype) -> SupportsFloat:
             if np.issubdtype(dtype, np.floating):
                 return np.finfo(dtype).precision
             else:
                 return np.inf
 
-        low_precision = _get_precision(self.low.dtype)
-        high_precision = _get_precision(self.high.dtype)
+        low_precision = _get_precision(low.dtype)
+        high_precision = _get_precision(high.dtype)
         dtype_precision = _get_precision(self.dtype)
-        if min(low_precision, high_precision) > dtype_precision:
+        if min(low_precision, high_precision) > dtype_precision:  # type: ignore
             logger.warn(f"Box bound precision lowered by casting to {self.dtype}")
-        self.low = self.low.astype(self.dtype)
-        self.high = self.high.astype(self.dtype)
+        self.low = low.astype(self.dtype)
+        self.high = high.astype(self.dtype)
 
         # Boolean arrays which indicate the interval type for each coordinate
         self.bounded_below = -np.inf < self.low
         self.bounded_above = np.inf > self.high
 
         super().__init__(self.shape, self.dtype, seed)
 
-    def is_bounded(self, manner="both"):
-        below = np.all(self.bounded_below)
-        above = np.all(self.bounded_above)
+    @property
+    def shape(self) -> Tuple[int, ...]:
```

Comment:
Is it possible to only add a type hint for that case, and not override the implementation? It's the same so ultimately it doesn't matter much, but it looks a bit odd.

My intuition is that only something like a type stub could do that which would be a bit of a hassle and not worth it in that case imo.

Reply:
Yeah I also believe type stubs are the only way to add types without touching the implementation.